### ISSUE TITLE
[BUG] compile pre-decomposes at most once

### ISSUE THREAD

**Issue by KetpuntoG** [PennyLaneAI/pennylane#4799]
### Expected behavior

I would expect that if I do qml.compile() of a template and its adjoint template, the decompositions would be equivalent but adjoined. However the depth of the decomposition is different:

<img width="300" alt="Captura de pantalla 2023-11-07 a las 13 58 38" src="https://github.com/PennyLaneAI/pennylane/assets/65235481/b6d7525e-e751-47ea-824e-d1bda5701f73">

<img width="300" alt="Captura de pantalla 2023-11-07 a las 13 58 49" src="https://github.com/PennyLaneAI/pennylane/assets/65235481/967a1e09-0203-4501-b0c2-3c2e345cc5d7">

As we can see in one case it expands the QFT and in another it does not. I would have liked that they were equivalent so in both cases it shows (QFT or QFT^t)

### Actual behavior

Explained above

### Source code

```shell
import pennylane as qml

dev = qml.device("default.qubit")

@qml.qnode(dev)
@qml.compile()
def circuit():
  #qml.QuantumPhaseEstimation(qml.PauliZ(0), estimation_wires = range(1,3))
  qml.adjoint(qml.QuantumPhaseEstimation)(qml.PauliZ(0), estimation_wires = range(1,3))
  return qml.state()

qml.draw_mpl(circuit)()
```

### System information

```shell
Name: PennyLane
Version: 0.33.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane-Lightning

Platform info:           Linux-5.15.120+-x86_64-with-glibc2.35
Python version:          3.10.12
Numpy version:           1.23.5
Scipy version:           1.11.3
Installed devices:
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by trbromley** [comment]
Thanks @KetpuntoG! Why do you prefer to not expand out the `QFT`? I agree with the issue but my expectation is that the QFT gets expanded out in both cases.

**Comment by trbromley** [comment]
Also, not that #4800 expands out the QFT @albi3ro.

**Comment by KetpuntoG** [comment]
My reason here to expand, is to see the operator inside, or because is more elegant the drawing 
In the case of QPE, I expand because a wanted to see the 3 main peaces.
(Perhaps using qml.compile() is not a good practice)

The other approach that could work is : 

<img width="872" alt="Captura de pantalla 2023-11-08 a las 10 59 16" src="https://github.com/PennyLaneAI/pennylane/assets/65235481/0a5e7039-dddf-4e3a-b4f2-75d6b1d4ddfd">

but here the operator is queued twice.
I know that I could do it outside the node and apply the decomposition ops but it's not very friendly 

Is there a simple way to do this?

(The particular case of this PR is that I was making an operator that included QPE and adjoint(QPE) and I wanted to paint it beautifully but each expansion came out different)

**Comment by KetpuntoG** [comment]
This flexibility to expand is something that Olivia also missed in PL and I think she was working on it in the Research Retreat (although this may be another topic)

**Comment by albi3ro** [comment]
I think for `compile`, the goal is to get to a fully broken down target gateset so that we can actually combine things. If left a `QFT` in the circuit before doing compilation, we couldn't actually be able to compile anything.

If you don't actually want to compile, I would recommend:

```
import pennylane as qml

dev = qml.device("default.qubit")

@partial(qml.devices.preprocess.decompose, stopping_condition = lambda obj: False, max_expansion=1)
@qml.qnode(dev)
def circuit():
  #qml.QuantumPhaseEstimation(qml.PauliZ(0), estimation_wires = range(1,3))
  qml.adjoint(qml.QuantumPhaseEstimation)(qml.PauliZ(0), estimation_wires = range(1,3))
  return qml.state()

qml.draw_mpl(circuit)()
```

Even though that transform is currently tailored toward device preprocessing.

**Comment by trbromley** [comment]
Yep, compile is more for swapping around gates to potentially minimize depth (transpile is another word for this) - in that case I agree with @albi3ro that we probably do want to expand out the QFT. Although, according to the code, QFT should not be expanded out because it is a fundamental op rather than a template :thinking: 

Thanks @KetpuntoG, we do recognize that better control over the decomposition pipeline is needed.

**Comment by albi3ro** [comment]
@KetpuntoG is this still an issue?

**Comment by albi3ro** [comment]
I'm going to close this as `decompose` is the tool for visualization.
